refactor: init CLI and @nuxtjs/color-mode module config

selemondev opened the pull request:
<!---☝️ PR title should follow conventional commits (https://conventionalcommits.org) -->

### 🔗 Linked issue

Fixes: #864

### ❓ Type of change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This pull request is intended to do two things namely:

1. **Configure the @nuxtjs/color-mode module by adding the `classSuffix: ''` property as shown below:** 
`await installModule('@nuxtjs/color-mode', { classSuffix: '' })`.

2. **Refactor the init CLI by ensuring that the `shadcn-nuxt` module is installed when the selected framework is Nuxt and the base dependencies as well.**


### 📸 Screenshots (if appropriate)

<!-- Add screenshots to help explain the change. -->

### 📝 Checklist

<!-- Put an `x` in all the boxes that apply. -->
<!-- If your change requires a documentation PR, please link it appropriately -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

selemondev committed:
refactor: init CLI

sadeghbarati reviewed (commented):
@selemondev Thanks for the PR

Please add `@nuxtjs/tailwindcss` and `@nuxtjs/color-mode` to the `shadcn-nuxt` package.json dependencies.
Remove TypeScript from dependencies maybe?


---

Could you please revert the CLI `init` changes? because there is a manual installation module way

We don't need to force user to use `shadcn-nuxt`

Related https://github.com/unovue/shadcn-vue/issues/883

selemondev commented:
Hey there @sadeghbarati, thanks for taking the time to review my PR and leave a comment 😁. Let me know what you think of these:

1. We let the developer choose their own CSS framework ( Tailwind or UnoCSS ). They will have to install it manually. This will solve issue #883 .

2. When the framework selected is `Nuxt`, we will install all the base dependencies ( radix-vue, class-variance-authority, tailwind-merge etc ) and the `shadcn-nuxt` module automatically. The `shadcn-nuxt` module will be configured automatically and contain the `@nuxtjs/color-mode` module and it's configuration.

zernonia commented:
Thanks for the PR @selemondev , at first the idea of the `shadcn-nuxt` module is to support component `autoImport`. Now that with this PR it will install `shadcn-nuxt`, which will install `@nuxtjs/tailwindcss`, `@nuxtjs/color-mode` automatically. Perhaps we can look into updating the [Nuxt Installation step](https://www.shadcn-vue.com/docs/installation/nuxt.html) ?

selemondev commented:
> Thanks for the PR @selemondev , at first the idea of the `shadcn-nuxt` module is to support component `autoImport`. Now that with this PR it will install `shadcn-nuxt`, which will install `@nuxtjs/tailwindcss`, `@nuxtjs/color-mode` automatically. Perhaps we can look into updating the [Nuxt Installation step](https://www.shadcn-vue.com/docs/installation/nuxt.html) ?

Hey @zernonia, I have updated the documentation. Here is what I have done:

1. I have removed the `@nuxtjs/tailwindcss` module installation process but kept it in the `manual` installation process.
2. The documentation changes walks the developers through the installation and configuration process for each module that is needed if they choose the manual installation process over installing the `shadcn-nuxt` module.

@sadeghbarati, for now we can work with automatically installing the `@nuxtjs/tailwindcss` module as TailwindCss is the default CSS framework used in Shadcn.

selemondev committed:
docs: update documentation with necessary changes

brokuka commented:
Any news?

zernonia commented:
Closing this as the cli is now irrelevant 🙏🏼
